20_beta_diversity.Rmd

```diff
@@ -247,6 +247,151 @@ As a final note, `mia` provides functions for the evaluation of additional dissi
 * `calculateOverlap`, `runOverlap` ()
 * `calculateDPCoA`, `runDPCoA` (Double Principal Coordinate Analysis)
 
+Redundancy analysis is similar to PCA, however, it takes into account covariates. 
+It aims to maximize the variance in respect of covariates. The results shows how much
+each covariate affects.
+
+```{r microbiome_RDA1}
+# Load required packages
+if(!require("vegan")){
+    install.packages("vegan")
+    library("vegan")
+}
+if(!require("stringr")){
+    install.packages("stringr")
+    library("stringr")
+}
+if(!require("knitr")){
+    install.packages("knitr")
+    library("knitr")
+}
+# Load data
+data(enterotype)
+# Covariates that are being analyzed
+variable_names <- c("ClinicalStatus", "Gender", "Age")
+
+# Apply relative transform
+enterotype <- transformSamples(enterotype, method = "relabundance")
+
+# Get assay
+assay <- t(assay(enterotype, "relabundance"))
+# Get colData
+coldata <- colData(enterotype)
+
+# Create a formula
+formula <- as.formula(paste0("assay ~ ", str_c(variable_names, collapse = " + ")) )
+
+# # Perform RDA
+rda <- rda(formula, data = coldata, scale = TRUE, na.action = na.exclude)
```

Comment:
Have you checked `mia::runRDA` and `mia::calculateRDA`? I wonder if they could do the same directly. May even use the same vegan functionality.

Reply:
Although, if direct calculation can give essential further information, then it might be a useful additional example. Or we could see how to develop runRDA/calculateRDA. The method is so common that it would be useful to automate as much as possible.

Reply:
I would indeed modify mia functions so that `na.action` can be passed.

If it cannot be done now, can you open an issue there and also refer to the relevant OMA section. Then we can merge this for now (after dealing with the other issues) so that we have something to get started.

Reply:
It wasn't actually a big thing. Here is PR https://github.com/microbiome/mia/pull/226

Although, I have to still check that it works properly

Reply:
(It casued other error, when different na.action was used, but that it was actrually working correctly)

Reply:
Perfect